src/platform/plugins/shared/discover/public/application/main/state_management/redux/selectors.ts

```diff
@@ -7,9 +7,15 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
+import { createSelector } from '@reduxjs/toolkit';
 import type { DiscoverInternalState } from './types';
 
 export const selectTab = (state: DiscoverInternalState, tabId: string) => state.tabs.byId[tabId];
 
-export const selectAllTabs = (state: DiscoverInternalState) =>
-  state.tabs.allIds.map((id) => selectTab(state, id));
+export const selectAllTabs = createSelector(
+  [
+    (state: DiscoverInternalState) => state.tabs.allIds,
+    (state: DiscoverInternalState) => state.tabs.byId,
+  ],
+  (allIds, byId) => allIds.map((id) => byId[id])
+);
```

Comment:
This change isn't directly related to the PR, but I noticed this selector causes unnecessary re-renders in consuming components since it was returning a new array reference each time, breaking the default React Redux memoization. The recommended solution is to use Reselect to create a memoized selector.